specs/_features/epbs/fork-choice.md

```diff
@@ -0,0 +1,172 @@
+# ePBS -- Fork Choice
+
+## Table of contents
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is the modification of the fork choice accompanying the ePBS upgrade.
+
+## Helpers
+
+### `verify_inclusion_list`
+*[New in ePBS]*
+
+```python
+def verify_inclusion_list(state: BeaconState, block: BeaconBlock, inclusion_list: InclusionList, execution_engine: ExecutionEngine) -> bool:
+    """
+    returns true if the inclusion list is valid. 
+    """
+    # Check that the inclusion list corresponds to the block proposer
+    signed_summary = inclusion_list.summary
+    proposer_index = signed_summary.message.proposer_index
+    assert block.proposer_index == proposer_index
+
+    # TODO: These checks will also be performed by the EL surely so we can probably remove them from here.
+    # Check the summary and transaction list lengths
+    summary = signed_summary.message.summary
+    assert len(summary) <= MAX_TRANSACTIONS_PER_INCLUSION_LIST
+    assert len(inclusion_list.transactions) == len(summary)
+
+    # TODO: These checks will also be performed by the EL surely so we can probably remove them from here.
+    # Check that the total gas limit is bounded
+    total_gas_limit = sum( entry.gas_limit for entry in summary)
+    assert total_gas_limit <= MAX_GAS_PER_INCLUSION_LIST
+
+    # Check that the signature is correct
+    # TODO: do we need a new domain?
+    signing_root = compute_signing_root(signed_summary.message, get_domain(state, DOMAIN_BEACON_PROPOSER))
+    proposer = state.validators[proposer_index]
+    assert bls.Verify(proposer.pubkey, signing_root, signed_summary.signature)
+   
+    # Check that the inclusion list is valid
+    return execution_engine.notify_new_inclusion_list(inclusion_list) 
```

Comment:
Maybe we should move `notify_new_inclusion_list` to `forkchoice.md`

Reply:
Engine methods are defined in beaconcha.in since Bellatrix. I don't oppose this. Just think it's too early to start discussing these issues when the spec is so early still